### PR TITLE
Add ipinfoio_facts module

### PR DESCRIPTION
**Pull request by akostyuk**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ipinfoio_facts

##### ANSIBLE VERSION

```
ansible 2.3.0 (devel fcb033b9e4) last updated 2016/11/15 21:42:17 (GMT +300)
  lib/ansible/modules/core: (detached HEAD a4cddac368) last updated 2016/11/15 21:42:43 (GMT +300)
  lib/ansible/modules/extras: (detached HEAD 19c7d5b31b) last updated 2016/11/15 21:42:44 (GMT +300)
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY

Retrieve geolocation data of a host's IP address using ipinfo.io API

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @akostyuk for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: network/ipinfoio_facts.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                    msg='Failed to parse the ipinfo.io response: '
+                    '{0} {1}'.format(self.url, content))
+            else:
+                return result
+
+
+def main():
+    module = AnsibleModule( # NOQA
+        argument_spec=dict(
+            http_agent=dict(default=USER_AGENT),
```
Comment (resmo) [inline network/ipinfoio_facts.py:122]: this arg spec is missing in the docs

Comment (akostyuk) [reply network/ipinfoio_facts.py:122]: @resmo thank you for pointing this out to me, I just forgot to include this option in docs.

File: network/ipinfoio_facts.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        supports_check_mode=True,
+    )
+
+    ipinfoio = IpinfoioFacts(module)
+    ipinfoio_result = dict(
+        changed=False, ansible_facts=ipinfoio.get_geo_data())
+    module.exit_json(**ipinfoio_result)
+
+from ansible.module_utils.basic import * # NOQA
+from ansible.module_utils.urls import * # NOQA
```
Comment (resmo) [inline network/ipinfoio_facts.py:134]: It has become best practice for modules to only import things you use e.g. and there is no advantage anymore to import them late in the code,  I would suggest move the import to above the class definition but below docs.

~~~python
from ansible.module_utils.basic import AnsibleModule
from ansible.module_utils.urls import fetch_url
~~~

**Review by resmo — APPROVED** [review]
Tested and works as expected.

**Comment by resmo** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @akostyuk for this PR. This PR was reviewed by an Ansible member. Marking for inclusion.

However, we would still appreciate if the maintainer(s) @ansible/core could look over it.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**